src/corehost/corehost.cpp

```diff
@@ -134,16 +140,95 @@ bool resolve_fxr_path(const pal::string_t& host_path, const pal::string_t& app_r
             app_root.c_str(),
             default_install_location.c_str(),
             dotnet_root_env_var_name.c_str());
-#else
-        trace::error(_X("A fatal error occurred, the folder [%s] does not exist"), fxr_dir.c_str()); 
-#endif
         return false;
     }
 
-    trace::info(_X("Reading fx resolver directory=[%s]"), fxr_dir.c_str());
+    if (!get_latest_fxr(std::move(fxr_dir), out_fxr_path))
+        return false;
+
+    return true;
+}
+
+#elif FEATURE_LIBHOST
+#define CURHOST_TYPE    _X("libhost")
+#define CUREXE_PKG_VER  LIBHOST_PKG_VER
+#define CURHOST_LIB
+
+bool resolve_fxr_path(const pal::string_t& root_path, pal::string_t* out_dotnet_root, pal::string_t* out_fxr_path)
+{
+    // If a hostfxr exists in root_path, then assume self-contained.
+    if (library_exists_in_dir(root_path, LIBFXR_NAME, out_fxr_path))
+    {
+        trace::info(_X("Resolved fxr [%s]..."), out_fxr_path->c_str());
+        out_dotnet_root->assign(root_path);
+        return true;
+    }
+
```

Comment:
What's the reason to not support `DOTNET_ROOT`?
I would expect the `hostfxr` search to be basically identical between apphost and libhost as they are both local to the app/component and both need to support framework dependent.
The only difference I could see us do is the check above - libhost + self-contained - not sure we want to support that right now (I don't think we thought it through...) - there's also the minor fact that SDK can't really produce that.

Reply:
I don't think supporting `DOTNET_ROOT` for COM hosting is the right thing to do purely because there is no need at the moment since this only works on Windows and our install locations are well-defined and the fact that it is yet another option that makes the support matrix even more complicated. In line with that, the work being done for version roll forward and how COM will always do major version should make it unnecessary. I can add it, but I don't see a strong need for it. I am in favor of making it relevant in the libhost scenario though.

Reply:
My view is about predictability. We already have feedback that it's "surprising" that `dotnet.exe` doesn't support `DOTNET_ROOT` while `apphost` does.
The other perspective is testability (even for our own tests) - `DOTNET_ROOT` is the only way to redirect comhost to a private copy of framework. apphost is kind of similar (although it does support self-contained).
IJW will be in the same boat.
We could add a "test-only" env variable to redirect, but given the apphost has it public, I would just use it everywhere.

Reply:
Boo. I abhor test-only env variables. Will bring back support.